lib/inspec/schema/schema.rb

```diff
@@ -0,0 +1,64 @@
+require "json"
+require "lib/inspec/schema/primitives"
+require "lib/inspec/schema/exec_json"
+require "lib/inspec/schema/exec_json_min"
+require "lib/inspec/schema/profile_json"
+
+module Inspec
+  module OutputSchema
+    # Build our definitions
+    def self.build_definitions(schema_type)
+      # Init definition hash
+      definitions = {}
+
+      # Fetch all dependencies
+      defined_types = schema_type.all_depends
+
+      # Add them all to the definition hash
+      defined_types.each do |type|
+        definitions[type.name] = type.body
+      end
+
+      # Nest under "definitions" label
+      {
+        "definitions" => definitions,
+      }
+    end
+
+    # Helper function to automatically bundle a type with its dependencies
+    def self.finalize(schema_type)
+      schema_type.body.merge(OutputSchema.build_definitions(schema_type))
+    end
+
+    # using a proc here so we can lazy load it when we need
+    # TODO: Modernize
+    PLATFORMS = lambda do
```

Comment:
Why the TODO? Modernize in what sense? Why isn't it just done the way you want?

Reply:
I think a few of these things are historical, as we were trying to be as non-invasive as possible will still achieving our initial goal. We'll go through and remove any such comments of actions to be taken and replace them with any notes of things that still need to be done.

Reply:
^ I don't know what modernize means here either haha.